### ISSUE TITLE
Version 20150226 executes only first test in suite via RestAPI

### ISSUE THREAD

**Issue by sbellus** [unclebob/fitnesse#688]
I upgrade Fitnesse from version 20140901 to 20150226 and xml output for suite is wrong.  

Only first test is executed via ?suite&format=xml rest api.

20150226:
``` xml
jenkins> curl --verbose --connect-timeout 30 --max-time 10800 http://10.21.1.200/FitNesse.XX.YY.ZZ\?suite\&format=xml
*   Trying 10.21.1.200...
* Connected to 10.21.1.200 (10.21.1.200) port 80 (#0)
> GET /FitNesse.XX.YY.ZZ?suite&format=xml HTTP/1.1
> User-Agent: curl/7.41.0
> Host: 10.21.1.200
> Accept: */*
>
< HTTP/1.1 200 OK
< X-FitNesse-Test-Id: 16
< Connection: close
< Server: FitNesse-v20150226
< Transfer-Encoding: chunked
< Content-Type: text/xml
<
<?xml version="1.0"?>
<testResults>
  <FitNesseVersion>v20150226</FitNesseVersion>
  <rootPath>FitNesse.XX.YY.ZZ</rootPath>
    <result>
    <counts>
      <right>0</right>
      <wrong>0</wrong>
      <ignores>0</ignores>
      <exceptions>0</exceptions>
    </counts>
    <runTimeInMillis>5435</runTimeInMillis>
    <relativePageName>SuiteSetUp</relativePageName>
    <pageHistoryLink>FitNesse.XX.SuiteSetUp?pageHistory&amp;resultDate=20150323095404</pageHistoryLink>
                <content>SOME CONTENT</content>
          </result>
    <result>
    <counts>
      <right>329</right>
      <wrong>0</wrong>
      <ignores>0</ignores>
      <exceptions>0</exceptions>
    </counts>
    <runTimeInMillis>33506</runTimeInMillis>
    <relativePageName>AA</relativePageName>
    <pageHistoryLink>FitNesse.XX.YY.ZZ.AA?pageHistory&amp;resultDate=20150323095410</pageHistoryLink>
                <content>SOME CONTENT</content>
          </result>
    <finalCounts>
    <right>2</right>
    <wrong>0</wrong>
    <ignores>0</ignores>
    <exceptions>0</exceptions>
  </finalCounts>
  <totalRunTimeInMillis>84539</totalRunTimeInMillis>
* Closing connection 0
</testResults>jenkins>
```

20140901
``` xml
jenkins> curl --verbose --connect-timeout 30 --max-time 10800 http://10.21.1.200/FitNesse.XX.YY.ZZ\?suite\&format=xml\&includehtml
*   Trying 10.21.1.200...
* Connected to 10.21.1.200 (10.21.1.200) port 80 (#0)
> GET /FitNesse.XX.YY.ZZ?suite&format=xml&includehtml HTTP/1.1
> User-Agent: curl/7.41.0
> Host: 10.21.1.200
> Accept: */*
>
< HTTP/1.1 200 OK
< Connection: close
< Server: FitNesse-v20140901
< Transfer-Encoding: chunked
< Content-Type: text/xml
<
<?xml version="1.0"?>
<testResults>
  <FitNesseVersion>v20140901</FitNesseVersion>
  <rootPath>FitNesse.XX.YY.ZZ</rootPath>
    <result>
    <counts>
      <right>0</right>
      <wrong>0</wrong>
      <ignores>0</ignores>
      <exceptions>0</exceptions>
    </counts>
    <runTimeInMillis>5492</runTimeInMillis>
    <relativePageName>SuiteSetUp</relativePageName>
    <pageHistoryLink>FitNesse.XX.SuiteSetUp?pageHistory&amp;resultDate=20150323103442</pageHistoryLink>
                <content>SOME CONTENT</content>
          </result>
    <result>
    <counts>
      <right>329</right>
      <wrong>0</wrong>
      <ignores>0</ignores>
      <exceptions>0</exceptions>
    </counts>
    <runTimeInMillis>33328</runTimeInMillis>
    <relativePageName>AA</relativePageName>
    <pageHistoryLink>FitNesse.XX.YY.ZZ.AA?pageHistory&amp;resultDate=20150323103448</pageHistoryLink>
                <content>SOME CONTENT</content>
          </result>
    <result>
    <counts>
      <right>658</right>
      <wrong>0</wrong>
      <ignores>0</ignores>
      <exceptions>0</exceptions>
    </counts>
    <runTimeInMillis>206111</runTimeInMillis>
    <relativePageName>BB</relativePageName>
    <pageHistoryLink>FitNesse.XX.YY.ZZ.BB?pageHistory&amp;resultDate=20150323103521</pageHistoryLink>
                <content>SOME CONTENT</content>
          </result>
    <result>
    <counts>
      <right>10</right>
      <wrong>0</wrong>
      <ignores>0</ignores>
      <exceptions>0</exceptions>
    </counts>
    <runTimeInMillis>9108</runTimeInMillis>
    <relativePageName>CC</relativePageName>
    <pageHistoryLink>FitNesse.XX.YY.ZZ.CC?pageHistory&amp;resultDate=20150323103848</pageHistoryLink>
                <content>SOME CONTENT</content>
          </result>
    <result>
    <counts>
      <right>47</right>
      <wrong>0</wrong>
      <ignores>0</ignores>
      <exceptions>0</exceptions>
    </counts>
    <runTimeInMillis>11095</runTimeInMillis>
    <relativePageName>DD</relativePageName>
    <pageHistoryLink>FitNesse.XX.YY.ZZ.DD?pageHistory&amp;resultDate=20150323103857</pageHistoryLink>
                <content>SOME CONTENT</content>
          </result>
    <result>
    <counts>
      <right>0</right>
      <wrong>0</wrong>
      <ignores>0</ignores>
      <exceptions>0</exceptions>
    </counts>
    <runTimeInMillis>885</runTimeInMillis>
    <relativePageName>SuiteTearDown</relativePageName>
    <pageHistoryLink>FitNesse.XX.SuiteTearDown?pageHistory&amp;resultDate=20150323103908</pageHistoryLink>
                <content>SOME CONTENT</content>
          </result>
    <finalCounts>
    <right>6</right>
    <wrong>0</wrong>
    <ignores>0</ignores>
    <exceptions>0</exceptions>
  </finalCounts>
  <totalRunTimeInMillis>267291</totalRunTimeInMillis>
* Closing connection 0
</testResults>jenkins>
```

**Comment by sbellus** [comment]
After some deeper investigation I found out that the problem is related to slim.timeout.
Up to now we don't configure timeout.
When I configure timeout to 600 (seconds) then I get xml for all tests in suite also with version 20150226.
What is strange for me that timeout was introduced in version 2014-04-18.
In which version FitNesse starts using this timeout?

**Comment by six42** [comment]
PR #660 was made to fix this.

**Comment by amolenaar** [comment]
Cool. The slim.timeout property has been implemented a little too strict in 20150226. It has already been fixed by #660.
